firewood/src/root_store.rs

```diff
@@ -108,3 +109,83 @@ impl RootStore for MockStore {
         Ok(self.roots.lock().expect("poisoned lock").get(hash).copied())
     }
 }
+
+use fjall::{Config, Keyspace, PartitionCreateOptions, PartitionHandle, PersistMode};
+
+pub struct FjallStore {
+    keyspace: Keyspace,
+    items: PartitionHandle,
+}
+
+impl Debug for FjallStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FjallStore")
+            .field("keyspace", &"<Keyspace>")
+            .field("items", &"<PartitionHandle>")
+            .finish()
+    }
+}
+
+impl FjallStore {
+    /// Creates or opens an instance of `FjallStore`.
+    ///
+    /// Args:
+    /// - path: the directory where `FjallStore` will write to.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if unable to create or open an instance of `FjallStore`.
+    pub fn new<P: AsRef<Path>>(path: P) -> Result<FjallStore, RootStoreError> {
+        let keyspace = Config::new(path).open().map_err(|e| RootStoreError {
+            method: RootStoreMethod::New,
+            source: Box::new(e),
+        })?;
+        let items = keyspace
+            .open_partition("firewood", PartitionCreateOptions::default())
```

Comment:
Let's pull "firewood" out as a const for better readability as to what this is used for:
```
const FJALL_PARTITION_NAME: &str = "firewood";
```